### PR TITLE
docs: import audit glossary from SEO doc

### PR DESCRIPTION
**Pull request by paulirish**
@rviscomi wrote some nice definitions (in a private doc) so i figured we should import them to the project.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -9,9 +9,14 @@ _Some incomplete notes_
 * **Driver** - Interfaces with [Chrome Debugging Protocol](https://developer.chrome.com/devtools/docs/debugger-protocol)  ([API viewer](https://chromedevtools.github.io/debugger-protocol-viewer/))
 * **Gatherers** - Uses Driver to collect information about the page. Minimal post-processing.
   * **Artifacts** - output of a gatherer
-* **Audits** - Using the Artifacts as input, Audits evaluate a test and assign pass/fail/scoring.
+* **Audit** - Tests for a single feature/optimization/metric. Using the Artifacts as input, an audit evaluates a test and  Resolves to a score, which may be pass/fail/numeric.  Meta info may contain markdown for rich text (code, links, etc).
```
Comment (patrickhulce) [inline docs/architecture.md]: nit: "and resolves to a score"

also do we support an etc for markdown or is it just code and links for now?

File: docs/architecture.md [hunk 2]
```diff
@@ -9,9 +9,14 @@ _Some incomplete notes_
...
   * **Computed Artifacts** - Generated on-demand from artifacts, these add additional meaning, and are often shared amongst multiple audits.
-* **Categories** - Grouping audit results into a user-facing section of the report (eg. `Best Practices`). Applies weighting and overall scoring to the section.
+  
+### Audit/Report terminology
+* **Audit Category** - Roll-up collection of audits and audit groups into a user-facing section of the report (eg. `Best Practices`). Applies weighting and overall scoring to the section. Examples: PWA, A11y, SEO.
```
Comment (patrickhulce) [inline docs/architecture.md]: do we want to use at least one example that isn't an acronym 😆

File: docs/architecture.md (hunk 1 above)
Comment (paulirish) [reply docs/architecture.md]: just code and links. code in title, links in description. and that's it, currently.

File: docs/architecture.md (hunk 2 above)
Comment (paulirish) [reply docs/architecture.md]: done

File: docs/architecture.md [hunk 3]
```diff
@@ -9,9 +9,14 @@ _Some incomplete notes_
...
 * **Gatherers** - Uses Driver to collect information about the page. Minimal post-processing.
   * **Artifacts** - output of a gatherer
-* **Audits** - Using the Artifacts as input, Audits evaluate a test and assign pass/fail/scoring.
+* **Audit** - Tests for a single feature/optimization/metric. Using the Artifacts as input, an audit evaluates a test and  resolves to a score which may be pass/fail/numeric. Formatting note: The meta description may contain markdown links and meta title may contain markdown code.
   * **Computed Artifacts** - Generated on-demand from artifacts, these add additional meaning, and are often shared amongst multiple audits.
-* **Categories** - Grouping audit results into a user-facing section of the report (eg. `Best Practices`). Applies weighting and overall scoring to the section.
+  
+### Audit/Report terminology
+* **Audit Category** - Roll-up collection of audits and audit groups into a user-facing section of the report (eg. `Best Practices`). Applies weighting and overall scoring to the section. Examples: PWA, Accessibility, Best Practices.
```
Comment (brendankenny) [inline docs/architecture.md]: we don't use the audit's `category` any more ([discussion](https://github.com/GoogleChrome/lighthouse/pull/3227#discussion_r137057437)), so this should maybe remain just `Categories` (as defined in the config), not `Audit Category`

Comment (paulirish) [reply docs/architecture.md]: sg

Comment (paulirish) [reply docs/architecture.md]: done

**Pull request merged**